test/utils/SchemaTester.js

```diff
@@ -0,0 +1,108 @@
+const testSchema = (Model) => ({
+    fieldRequired: (fieldName, customObject) => {
+        test(`'${fieldName}' is required`, async () => {
+            const emptyObject = new Model(customObject || {});
+            try {
+                await emptyObject.validate();
+            } catch (err) {
+                expect(err.errors[fieldName]).toBeDefined();
+                expect(err.errors[fieldName]).toHaveProperty("kind", "required");
+                expect(err.errors[fieldName]).toHaveProperty("message", `Path \`${fieldName}\` is required.`);
+            }
+        });
+    },
+
+    minLength: (fieldName, minimum) => {
+        describe(`${fieldName} must have a minimum of ${minimum} characters`, () => {
+            test("check for error", async () => {
+                const modelObj = {};
+                modelObj[fieldName] = "a".repeat(minimum - 1);
+
+                const model = new Model(modelObj);
+                try {
+                    await model.validate();
+                } catch (err) {
+                    expect(err.errors[fieldName]).toBeDefined();
+                    expect(err.errors[fieldName]).toHaveProperty("kind", "minlength");
+                    expect(err.errors[fieldName]).toHaveProperty("message",
+                        `Path \`${fieldName}\` (\`${"a".repeat(minimum - 1)}\`) is shorter than the minimum allowed length (${minimum}).`
+                    );
+                }
+            });
+            test("check for success", async () => {
```

Comment:
```suggestion
            test("at minimum should not throw error", async () => {
```

I think this might be a clearer name? Not super sure about this one